clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -638,13 +638,27 @@ public void testQuotaMetrics() throws Exception {
 
     @Test
     public void testGetOffsetsByTimes() {
-        client.prepareResponseFrom(listOffsetResponse(Errors.NONE, 100L, 100L), cluster.leaderFor(tp));
+        TopicPartition tp0 = tp;
+        TopicPartition tp1 = new TopicPartition(topicName, 1);
+        // Ensure metadata has both partition.
+        Cluster cluster = TestUtils.clusterWith(2, topicName, 2);
+        metadata.update(cluster, time.milliseconds());
 
-        Map<TopicPartition, OffsetAndTimestamp> offsetAndTimestampMap =
-                fetcher.getOffsetsByTimes(Collections.singletonMap(tp, 0L));
-        assertEquals(offsetAndTimestampMap.get(tp).timestamp(), 100L);
-        assertEquals(offsetAndTimestampMap.get(tp).offset(), 100L);
+        // First try should fail due to metadata error.
+        client.prepareResponseFrom(listOffsetResponse(Errors.NONE, 10L, 10L), cluster.leaderFor(tp0));
+        client.prepareResponseFrom(listOffsetResponse(tp1, Errors.NOT_LEADER_FOR_PARTITION, -1L, -1L), cluster.leaderFor(tp1));
```

Comment:
Also, in this case, the error is in the final response. Maybe we should have a case where the error response comes first?